### PR TITLE
[CFM-1476] Broken document widget

### PR DESCRIPTION
**Pull request by ybaras**
[CFM-1476](https://issuetracker.amplexor.com/jira/browse/CFM-1476)

### CODE REVIEW

**Comment by ybaras** [comment]
cfm-1476

Attaching a document to a new event via
# Browse, 
# Drag & drop, 
# Choose from library.

**Comment by ordavidil** [comment]
@ybaras 
please verify this functionality on IE 11 as well, thanks

**Comment by ordavidil** [comment]
and also by other users (not only _admin_

**Comment by ybaras** [comment]
@ordavidil Tested OK on Win 8.1 - IE 11 and with other users.

File: project/profiles/capacity4more/modules/c4m/restful/c4m_restful/plugins/restful/C4mRestfulEntityBaseNode.class.php [hunk 1]
```diff
@@ -18,7 +18,7 @@ class C4mRestfulEntityBaseNode extends RestfulEntityBaseNode {
    */
   protected function checkEntityAccess($op, $entity_type, $entity) {
     $account = $this->getAccount();
-    $resource_name = $this->getResourceName();
+    $resource_name = rtrim($this->getResourceName(), 's');
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/restful/c4m_restful/plugins/restful/C4mRestfulEntityBaseNode.class.php]:
```php
$this->getPluginKey('bundle');
```

File: project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module [hunk 2]
```diff
@@ -56,6 +56,26 @@ function c4m_og_current_group_id() {
 }
 
 /**
+ * Helper to get the current active group purl (based on og_context()).
+ *
+ * @return null|string
+ *   The group purl, or NULL if no context was found.
+ */
+function c4m_og_current_group_purl() {
+  $context = og_context();
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module]: we should use `c4m_og_current_group()` for most of the logic

File: project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module [hunk 3]
```diff
@@ -56,6 +56,26 @@ function c4m_og_current_group_id() {
 }
 
 /**
+ * Helper to get the current active group purl (based on og_context()).
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module]: fix the docs

File: project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module [hunk 4]
```diff
@@ -56,6 +56,26 @@ function c4m_og_current_group_id() {
...
+ */
+function c4m_og_current_group_purl() {
+  $context = og_context();
+  if (!$context) {
+    return NULL;
+  }
+
+  $group = node_load($context['gid']);
+  if (!$group) {
+    return FALSE;
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module]: return empty string

File: project/profiles/capacity4more/modules/c4m/widgets/c4m_document_widget/templates/c4m_document_widget.tpl.php [hunk 5]
```diff
@@ -18,7 +18,7 @@
         <p class="cfm-file-upload__title"><?php print t('Drop file here to upload documents'); ?></p>
         <a href="javascript://" ng-click="browseFiles('<?php print $field_name; ?>')"> <?php print t('or browse'); ?> </a>
       </div>
-      <a href="JavaScript://" onclick="Drupal.media.popups.mediaBrowser(function(media){window.location = '#overlay=nobelprize/overlay-file/' + media[0].fid;},{'enabledPlugins[media_default--media_browser_my_files]': 'media_default--media_browser_my_files'})"
+      <a href="JavaScript://" onclick="Drupal.media.popups.mediaBrowser(function(media){window.location = '#overlay=<?php print (c4m_og_current_group_purl()); ?>/overlay-file/' + media[0].fid;},{'enabledPlugins[media_default--media_browser_my_files]': 'media_default--media_browser_my_files'})"
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/widgets/c4m_document_widget/templates/c4m_document_widget.tpl.php]: move to `<script>`

File: project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module [hunk 6]
```diff
@@ -58,21 +58,11 @@ function c4m_og_current_group_id() {
...
-    return NULL;
-  }
-
-  $group = node_load($context['gid']);
-  if (!$group) {
-    return FALSE;
-  }
-
-  return $group->purl;
+  return ($group = c4m_og_current_group()) ? $group->purl:'';
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module]: let's use the standard `if` condition.

File: project/profiles/capacity4more/modules/c4m/widgets/c4m_document_widget/templates/c4m_document_widget.tpl.php [hunk 7]
```diff
@@ -18,7 +18,14 @@
         <p class="cfm-file-upload__title"><?php print t('Drop file here to upload documents'); ?></p>
         <a href="javascript://" ng-click="browseFiles('<?php print $field_name; ?>')"> <?php print t('or browse'); ?> </a>
       </div>
-      <a href="JavaScript://" onclick="Drupal.media.popups.mediaBrowser(function(media){window.location = '#overlay=<?php print (c4m_og_current_group_purl()); ?>/overlay-file/' + media[0].fid;},{'enabledPlugins[media_default--media_browser_my_files]': 'media_default--media_browser_my_files'})"
+      <script>
+        function c4m_attachment_overlay(media) {
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/widgets/c4m_document_widget/templates/c4m_document_widget.tpl.php]:
```js
var c4m_attachment_overlay = function (media) {
```

File: project/profiles/capacity4more/modules/c4m/widgets/c4m_document_widget/templates/c4m_document_widget.tpl.php [hunk 8]
```diff
@@ -18,7 +18,14 @@
...
+          window.location = '#overlay=<?php print (c4m_og_current_group_purl()); ?>/overlay-file/' + media[0].fid;
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/widgets/c4m_document_widget/templates/c4m_document_widget.tpl.php]: remove the extra parenthesis from the php function

**Comment by ordavidil** [comment]
@ybaras 
it seems that under projects it's not working properly...

**Comment by ybaras** [comment]
@ordavidil I tested every role in Chrome, IE, FF. No problems found.

**Pull request merged**